### ISSUE TITLE
Make REST API use Twenty ORM direclty

### ISSUE THREAD

**Issue by FelixMalfait** [twentyhq/twenty#3644]
**Objective**: Enhance the reliability and performance of the REST API.

## Technical Context
Currently, our REST API (documentation [here](https://twenty.com/developers/rest-api/core#)) is exposed via two dynamic endpoint sets:
- `rest/...`
- `rest/batch/...`
These endpoints are handled by two controllers: `rest-api-core.controller` and `rest-api-batch-core.controller`, both backed by `RestApiCoreService`.

The RestApiCoreService currently parses REST payloads, converts them into GraphQL payloads, and calls the GraphQL API—an inefficient process. This approach was initially necessary because querying workspaceSchemas in Twenty required pg_graphql.

However, we recently introduced TwentyORM (soon to be renamed WorkspaceORM), a lightweight layer on top of TypeORM, which can directly query workspaceSchemas.

Goal: Refactor the REST API to leverage TwentyORM directly.

Note: The GraphQL API is also being migrated to TwentyORM (replacing `pg_graphql`), so you can draw inspiration from the `GraphqlQueryRunnerService`.

## Key Goals:
Refactor the REST API to use TwentyORM directly.
Ensure the REST API is fully covered by integration tests.

## Technical inputs

We should take a lot of inspiration from what has been done in the new GraphqlQueryRunnerService. Advices:
1) split the code in services / utils to keep small and well scoped functional pieces
2) Overall logic: a) parse REST payload into TypeORM format, b) send TypeORM query, c) parse result back into REST response
3) All the pieces are already there: the current code is able to parse REST into graphql and Graphql into twentyORM, we "only" need to skip one step. It will likely mean to write everything from scratch but this should still speed a lot the process

**Comment by charlesBochet** [comment]
Yes please do!

If any experienced contributor wants to tackle this task, I'm happy to give a hand here. There is a lot of work but the engineering and the product is very well defined so it should be doable by the community

**Comment by charlesBochet** [comment]
@FelixMalfait It's not that difficult anymore I think as most of the complexity is within twenty-orm and we have quite good example of parsing in the current rest api and in our graphql implementation (we know how to parse rest input to graphql and graphql to orm already, we need to do rest input to orm direclty)

**Comment by charlesBochet** [comment]
I'm editing the description to add more details

**Comment by Faisal-imtiyaz123** [comment]
@charlesBochet I am ready to work. You may please assign me.

**Comment by FelixMalfait** [comment]
@Faisal-imtiyaz123 let me know if we should unassign you. Thanks!

**Comment by Faisal-imtiyaz123** [comment]
@FelixMalfait Yes you may unassign me. I talked to @charlesBochet regarding this.

**Comment by charlesBochet** [comment]
/oss.gg 3000

**Comment by DeepaPrasanna** [comment]
I have working experience in nestjs, May I work on this? @charlesBochet

**Comment by charlesBochet** [comment]
sure, thank you @DeepaPrasanna!
You can type: /assign to get assigned to it

**Comment by charlesBochet** [comment]
It's not an easy one, feel free to ping me if you need help!

**Comment by DeepaPrasanna** [comment]
sure! Thank you

**Comment by DeepaPrasanna** [comment]
hello @charlesBochet , I am working on this. I have an interview scheduled for today, that is why I paused for some time. I plan to resume my work again tomorrow.

**Comment by DeepaPrasanna** [comment]
I will be creating a draft PR today. Pls don't unassign me.

**Comment by FelixMalfait** [comment]
Thanks @DeepaPrasanna, great to create a PR to get intermediate feedback and make sure you go in the right direction since this is a big PR, thanks a lot!

**Comment by DeepaPrasanna** [comment]
Hello @charlesBochet @FelixMalfait , I have tried to refactor the `rest/batch/...` endpoint first. I have raised a draft PR. If I am moving in the right direction, I will refactor the remaining APIs soon. Requesting feedback :) Thank u

**Comment by charlesBochet** [comment]
Thank you @DeepaPrasanna, I will take a look today

**Comment by DeepaPrasanna** [comment]
Just curious, how the distribution of the points will work if I am unable to complete it everything before 31st. I do understand and respect your time for reviews. :)

**Comment by charlesBochet** [comment]
Awarded points on the PR :) We will take a deep look at your PR shortly, we are a bit overwhelmed by the volume of PRs right now!

**Comment by Nabhag8848** [comment]
@FelixMalfait @charlesBochet this is interesting. working on it, getting back with first version to get initial feedback on implementation, will start by making it work for batch endpoints as we will get enough idea for other endpoints with that implementation.

**Comment by DeepaPrasanna** [comment]
Hey @Nabhag8848, I wasn't able to finish the PR due to my work commitments. You can refer to the incomplete PR to get a head start.

**Comment by FelixMalfait** [comment]
Thanks a lot @Nabhag8848 this is a big one and an impactful one for the overall quality of the codebase (current solution for the api is hacky)! Ping us when you need feedback!

**Comment by Nabhag8848** [comment]
still working, have better idea about overall codebase 🙌🏼.

**Comment by martmull** [comment]
Hey @Nabhag8848 are you still working on that issue? @pacyL2K19 is interested to work on it

**Comment by pacyL2K19** [comment]
Can you assign this task to me @martmull?
The first PR is under review and I will be working on the second PR for the `POST /rest/*` endpoint from tomorrow

**Comment by martmull** [comment]
Done

**Comment by FelixMalfait** [comment]
Closing as I think this was broken down into subissues @martmull ? Great work @pacyL2K19 thanks again!
